### DISCUSSION TITLE
How to cleanup object after being borrow_mut ??

### DISCUSSION THREAD

**Discussion by ajaythxkur** [aptos-labs/aptos-developer-discussions#431] [Questions] [answered]
i want to delete object on a specific if condition:
The code goes like this
1. let object = borrow_mut object and change some state in it.
2. An if condition is specified in function, if the condition is achieved i want to delete the object with delete ref.

I can't delete the object with object.delete_ref

**Comment by MoonShiesty** [comment]
you need to remove all the resources inside of the object before you can destroy it, ie:
```
move_from<MyType>(object_addr);
```
then you can cleanup the object with the delete ref. if the delete ref is stored in the resource you can do this:
```
let MyType {
     delete_ref,
     ..., // other fields
} = move_from<MyType>(object_addr);
object::delete(delete_ref);
```

**Reply by kshitijc1506** [reply]
I don't think you _need_ to remove the resources? Although, it's probably better practice to do so, so the resources aren't in some weird dangling state.

**Reply by magnum6actual** [reply]
That would depend on the abilities of the resources, though, right? All would need 'drop' ability if not handled explicitly.

**Reply by kshitijc1506** [reply]
@magnum6actual you need to define the fields to be empty, which effectively "removes" the resources from global storage. Here's what `object::delete` does:

```rust
let object_core = move_from<ObjectCore>(ref.self);
let ObjectCore {
   guid_creation_num: _,
   owner: _,
   allow_ungated_transfer: _,
   transfer_events,
} = object_core;
```

**Reply by ajaythxkur** [reply]
let me explain with code what i am trying to achieve
fun(object: Object<ObjectCore>) acquires SomeStruct {
     let some_struct = borrow_global_mut<SomeStruct>(obj_addr);
     some_struct.counter = some_struct.counter + 1;
     if(some_struct.counter == 5){
     delete object here (can't move from as its mut borrow above)
     }
}

**Reply by MoonShiesty** [reply, marked as answer]
you should be able to borrow within a local scope:

ie: 
```
let isFive = false;
{
    / / borrowed within bracket's local scope
    let some_struct = borrow_global_mut(obj_addr);
    some_struct.counter = some_struct.counter + 1;
    isFive = some_struct.counter == 5;
}
if(isFive){
delete object here (previous borrow is out-of-scope)
```
